src/functions.jl

```diff
@@ -110,3 +97,33 @@ function Base.append!(A::NamedDimsArray{L,T,1}, B::AbstractVector) where {L,T}
     data = append!(parent(A), unname(B))
     return NamedDimsArray{newL}(data)
 end
+
+################################################
+# Generators
+
+if VERSION > v"1.1-"
+    Base.eachslice(A::NamedDimsArray; dims) = _eachslice(A, dims)
+else
+    eachcol(A::AbstractVecOrMat) = (view(A, :, i) for i in axes(A, 2))
+    eachrow(A::AbstractVecOrMat) = (view(A, i, :) for i in axes(A, 1))
+    # every line identical to Base, but no _eachslice(A, dims) to disatch on.
+    eachslice(A::AbstractArray; dims) = _eachslice(A, dims)
+end
+
+function _eachslice(A::AbstractArray, dims::Symbol)
+    numerical_dims = dim(A, dims)
+    return _eachslice(A, numerical_dims)
+end
+function _eachslice(A::AbstractArray, dims::Tuple)
+    length(dims) == 1 || throw(ArgumentError("only single dimensions are supported"))
+    return _eachslice(A, first(dims))
+end
+@inline function _eachslice(A::AbstractArray, dim::Int)
+    dim <= ndims(A) || throw(DimensionMismatch("A doesn't have $dim dimensions"))
+    idx1, idx2 = ntuple(d->(:), dim-1), ntuple(d->(:), ndims(A)-dim)
```

Comment:
This shoujld be on two seperate lines
I would title it `inds_before` and `inds_after`

Reply:
While I agree that that would be clearer, on its own, I think staying line-for-line identical to Base is a good idea if copying.